app/controllers/reports/regions_controller.rb

```diff
@@ -136,6 +136,20 @@ def monthly_district_data_report
     end
   end
 
+  def monthly_state_data_report
+    @region ||= authorize { current_admin.accessible_state_regions(:view_reports).find_by!(slug: report_params[:id]) }
+    @period = Period.month(params[:period] || Date.current)
+    csv = MonthlyStateDataService.new(@region, @period).report
+    report_date = @period.to_s.downcase
+    filename = "monthly-state-data-#{@region.slug}-#{report_date}.csv"
```

Comment:
The file name should be `monthly-district-data-#{@region.slug}-#{report_date}.csv`